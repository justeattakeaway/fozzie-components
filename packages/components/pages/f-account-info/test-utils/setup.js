import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
