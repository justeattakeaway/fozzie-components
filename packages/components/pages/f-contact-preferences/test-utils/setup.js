export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
