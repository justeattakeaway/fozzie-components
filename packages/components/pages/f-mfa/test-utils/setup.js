export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
