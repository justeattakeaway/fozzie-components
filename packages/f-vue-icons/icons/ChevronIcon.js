import _mergeJSXProps from "babel-helper-vue-jsx-merge-props";
export default {
  name: 'ChevronIcon',
  props: {},
  functional: true,
  render: function render(h, ctx) {
    var attrs = ctx.data.attrs || {};
    ctx.data.attrs = attrs;
    return h("svg", _mergeJSXProps([{
      attrs: {
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 10 6"
      },
      "class": "c-ficon c-ficon--chevron"
    }, ctx.data]), [h("path", {
      attrs: {
        d: "M5 4.41L9.29.12a.42.42 0 0 1 .59.6L5.29 5.28a.42.42 0 0 1-.58 0L.12.71a.42.42 0 0 1 .6-.59L5 4.42z",
        fill: "#333"
      }
    })]);
  }
};