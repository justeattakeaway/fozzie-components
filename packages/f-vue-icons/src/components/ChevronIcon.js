export default {
    name: 'ChevronIcon',

    props: {
    },

    functional: true,

    render (h, ctx) {
        const attrs = ctx.data.attrs || {};
        ctx.data.attrs = attrs;

        return <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 6" class="c-ficon c-ficon--chevron" {...ctx.data}><path d="M5 4.41L9.29.12a.42.42 0 0 1 .59.6L5.29 5.28a.42.42 0 0 1-.58 0L.12.71a.42.42 0 0 1 .6-.59L5 4.42z" fill="#333"></path></svg>;
    }
};
