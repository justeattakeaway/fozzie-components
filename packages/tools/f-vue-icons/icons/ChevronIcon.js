import _mergeJSXProps from "babel-helper-vue-jsx-merge-props";
export default {
  name: 'ChevronIcon',
  props: {},
  functional: true,
  render: function render(h, ctx) {
    var attrs = ctx.data.attrs || {};
    ctx.data.attrs = attrs;
    return h("svg", _mergeJSXProps([{
      attrs: {
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 11.171582 6.5885787"
      },
      "class": "c-ficon c-ficon--chevron"
    }, ctx.data]), [h("path", {
      attrs: {
        "fill-rule": "evenodd",
        d: "m 0.29029267,0.29828979 a 1,1 0 0 1 1.41699993,0 l 3.876,3.87400011 3.8769996,-3.87800011 a 1,1 0 0 1 1.4170008,0 1,1 0 0 1 0,1.41700011 L 6.2942926,6.2942896 a 1,1 0 0 1 -1.417,0 L 0.29429267,1.7112899 a 1,1 0 0 1 -0.004,-1.41300011 z"
      }
    })]);
  }
};