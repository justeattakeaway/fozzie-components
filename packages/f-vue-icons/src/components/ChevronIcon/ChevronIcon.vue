<template>
    <chevron-icon
        :class="['c-icon c-icon--chevron', {
            'c-icon--chevron--small': isSmall,
            'c-icon--chevron--up': isFacingUp,
            'c-icon--chevron--right': isFacingRight,
            ...iconClasses
        }]" />
</template>

<script>
import ChevronIcon from '@justeat/f-icons/src/img/icons/arrows/chevron.svg';
import iconsMixin from '../../mixins/icons.mixin';

export default {
    components: {
        ChevronIcon
    },

    mixins: [iconsMixin],

    props: {
        isSmall: {
            type: Boolean,
            default: false
        },

        isFacingUp: {
            type: Boolean,
            default: false
        },

        isFacingRight: {
            type: Boolean,
            default: false
        }
    }
};
</script>
