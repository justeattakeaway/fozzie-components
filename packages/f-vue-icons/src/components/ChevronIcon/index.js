import ChevronIcon from './ChevronIcon.vue';

const install = Vue => {
    Vue.component(ChevronIcon.name, ChevronIcon);
};

export default install;
