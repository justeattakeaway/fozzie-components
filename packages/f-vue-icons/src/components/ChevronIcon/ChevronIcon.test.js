import { shallowMount } from '@vue/test-utils';
import ChevronIcon from './ChevronIcon.vue';

describe('ChevronIcon', () => {
    it('should be defined', () => {
        const wrapper = shallowMount(ChevronIcon);
        expect(wrapper).toBeDefined();
    });

    it('should add component class by default', () => {
        // Arrange
        const wrapper = shallowMount(ChevronIcon);

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toEqual(['c-icon', 'c-icon--chevron']);
    });

    it('should render default component markup', () => {
        // Arrange & Act
        const wrapper = shallowMount(ChevronIcon);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });

    it('should add small class', () => {
        // Arrange
        const propsData = { isSmall: true };
        const wrapper = shallowMount(ChevronIcon, {
            propsData
        });

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toContain('c-icon--chevron--small');
    });

    it('should add up class', () => {
        // Arrange
        const propsData = { isFacingUp: true };
        const wrapper = shallowMount(ChevronIcon, {
            propsData
        });

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toContain('c-icon--chevron--up');
    });

    it('should add left class', () => {
        // Arrange
        const propsData = { isFacingLeft: true };
        const wrapper = shallowMount(ChevronIcon, {
            propsData
        });

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toContain('c-icon--chevron--left');
    });

    it('should add right class', () => {
        // Arrange
        const propsData = { isFacingRight: true };
        const wrapper = shallowMount(ChevronIcon, {
            propsData
        });

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toContain('c-icon--chevron--right');
    });

    it('should be able to add multiple classes', () => {
        // Arrange
        const propsData = {
            isSmall: true,
            isFacingUp: true
        };
        const wrapper = shallowMount(ChevronIcon, {
            propsData
        });

        // Act
        const result = wrapper.classes();

        // Assert
        expect(result).toEqual(['c-icon', 'c-icon--chevron', 'c-icon--chevron--small', 'c-icon--chevron--up']);
    });
});
