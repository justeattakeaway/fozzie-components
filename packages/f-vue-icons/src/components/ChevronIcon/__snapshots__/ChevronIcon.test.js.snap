// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronIcon should render default component markup 1`] = `<chevron-icon-stub class="c-icon c-icon--chevron"></chevron-icon-stub>`;
